flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -373,7 +383,15 @@ public <N> void setCurrentNamespaceForState(
     }
 
     <IN, OUT> void insertActiveBuffer(StateRequest<K, ?, IN, OUT> request) {
-        stateRequestsBuffer.enqueueToActive(request);
+        if (request.isSync()) {
+            if (request.getRequestType() == StateRequestType.SYNC_POINT) {
+                request.getFuture().complete(null);
+            } else {
+                stateExecutor.executeRequest(request);
+            }
+        } else {
+            stateRequestsBuffer.enqueueToActive(request);
```

Comment:
If `!request.isSync()`, will the request be `SYNC_POINT` here?
Maybe some test call handleRequest(SYNC_POINT) directly.

Reply:
Nope, `SYNC_POINT` is always `sync = true` (see [here](https://github.com/apache/flink/pull/26005/files/7b135345904955fed266dae61b372f22c2acf745#diff-916a2a7c5aa87e2de74a843ba43f6553bf3d14a6d4cb7cf3b76b4a562cd7a0e0R333)). I was considering introduce `sync || type == SYNC_POINT` in the constructor of `StateRequest`, but the execution is more up to the `AEC`, so I keep it there.